ui/protocol.go

```diff
@@ -0,0 +1,97 @@
+package ui
+
+import (
+	"github.com/algorandfoundation/hack-tui/internal"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+	"strconv"
+	"strings"
+)
+
+// ProtocolViewModel includes the internal.StatusModel and internal.MetricsModel
+type ProtocolViewModel struct {
+	tea.Model
+	Status    *internal.StatusModel
+	Metrics   *internal.MetricsModel
+	ViewWidth int
+	IsVisible bool
+}
+
+// Init has no I/O right now
+func (m ProtocolViewModel) Init() tea.Cmd {
+	return nil
+}
+
+// Update applies a message to the model and returns an updated model and command.
+func (m ProtocolViewModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	return m.HandleMessage(msg)
+}
+
+// HandleMessage processes incoming messages and updates the ProtocolViewModel's state.
+// It handles tea.WindowSizeMsg to update ViewWidth and tea.KeyMsg for key events like 'h' to toggle visibility and 'q' or 'ctrl+c' to quit.
+func (m ProtocolViewModel) HandleMessage(msg tea.Msg) (ProtocolViewModel, tea.Cmd) {
+	switch msg := msg.(type) {
+	// Update Viewport Size
+	case tea.WindowSizeMsg:
+		m.ViewWidth = msg.Width
+	case tea.KeyMsg:
+		switch msg.String() {
+		// The H key should hide the render
+		case "h":
+			m.IsVisible = !m.IsVisible
+		case "q", "ctrl+c":
+			return m, tea.Quit
+		}
+	}
+	// Return the updated model to the Bubble Tea runtime for processing.
+	return m, nil
+}
+
+// View renders the view for the ProtocolViewModel according to the current state and dimensions.
+func (m ProtocolViewModel) View() string {
+	if !m.IsVisible {
+		return ""
+	}
+	if m.ViewWidth <= 0 {
+		return "Loading...\n\n\n\n\n\n"
+	}
+	beginning := blue.Render(" Node: ") + m.Status.Version
+	end := ""
+	if m.Status.NeedsUpdate {
+		end = green.Render("[UPDATE AVAILABLE] ")
+	}
+
+	middle := strings.Repeat(" ", max(0, m.ViewWidth/2-(lipgloss.Width(beginning)+lipgloss.Width(end)+2)))
+
+	// Last Round
+	row1 := lipgloss.JoinHorizontal(lipgloss.Left, beginning, middle, end)
+
+	row2 := blue.Render(" Network: ") + m.Status.Network
+
+	row3 := blue.Render(" Protocol Voting: ") + strconv.FormatBool(m.Status.Voting)
+
+	return topSections(max(0, m.ViewWidth/2)).Render(lipgloss.JoinVertical(lipgloss.Left,
+		row1,
+		"",
+		row2,
+		"",
+		row3,
+	))
+}
+
+// MakeProtocolViewModel constructs a ProtocolViewModel using a given StatusModel and predefined metrics.
+func MakeProtocolViewModel(status *internal.StatusModel) ProtocolViewModel {
+	metrics := internal.MetricsModel{
+		RoundTime: 2.87,
```

Comment:
I'm not seeing these values in the Status View. Latest round (m.Status.LastRound) is being shown, but m.Metrics.RoundTime, m.Metrics.TPS, m.Metrics.RX and m.Metrics.TX are 0 right now.

Reply:
<img width="584" alt="Screenshot 2024-10-17 at 11 59 55" src="https://github.com/user-attachments/assets/77cb7c8b-677d-48d4-8844-b294b2285fb2">

Reply:
I don't see much benefit of changing it considering that it's going to be refactored when it's actually integrated.